script/playbackstrategy/msestrategy.js

```diff
@@ -73,12 +85,30 @@ define('bigscreenplayer/playbackstrategy/msestrategy',
       function onQualityChangeRendered (event) {
         if (event.mediaType === 'video') {
           DebugTool.info('ABR Change Rendered from: ' + event.oldQuality + ' to: ' + event.newQuality);
+
+          if (!bitrateInfoList) {
+            bitrateInfoList = mediaPlayer.getBitrateInfoListFor('video');
+          }
+          if (bitrateInfoList && event.newQuality) {
+            playerMetadata.playbackBitrate = bitrateInfoList[event.newQuality].bitrate / 1000;
+          }
+          Plugins.interface.onPlayerInfoUpdated(playerMetadata);
         }
       }
 
       function onMetricAdded (event) {
-        if (event.mediaType === 'video' && event.metric === 'DroppedFrames') {
-          DebugTool.keyValue({key: 'Dropped Frames', value: event.value.droppedFrames});
+        if (event.mediaType === 'video') {
+          if (event.metric === 'DroppedFrames') {
+            DebugTool.keyValue({key: 'Dropped Frames', value: event.value.droppedFrames});
+          } else if (event.metric === 'BufferLevel') {
+            videoMetrics = mediaPlayer.getMetricsFor('video');
+            dashMetrics = mediaPlayer.getDashMetrics();
+
+            if (videoMetrics && dashMetrics) {
+              playerMetadata.bufferLength = dashMetrics.getCurrentBufferLevel(videoMetrics);
```

Comment:
Maybe match the naming up here? buffer length -> buffer level

Reply:
I guess rDot wants bufferLength, so there's a mismatch somewhere. I think here is ok.